software/edge/src/aeris_orchestrator/test/test_mission_node.py

```
@@ -699,6 +747,305 @@ def test_detection_selection_falls_back_to_most_recent_seen_without_positions(
+        def update_setpoint(self, setpoint: dict[str, float]) -> None:
+            calls["update"].append(dict(setpoint))
+
+        def close(self) -> None:
+            self.is_streaming = False
+            calls["close"].append(True)
+
+    fake_adapter = _FakeRangerAdapter()
+    monkeypatch.setattr(
+        MissionNode,
+        "_build_ranger_adapter",
+        lambda self, endpoint: fake_adapter,
```

Ruff (0.14.14)

[warning] Unused lambda argument: `self`

(ARG005)